cmd/create/create_agent.go

```diff
@@ -0,0 +1,439 @@
+	Namespace string
+
+	Context context.Context
+
+	ArmoryClient      *configuration.ConfigClient
+	configuration     *config.Configuration
+	contextNames      []string
+	credentials       *model.Credential
+	kubernetesFactory cmdutil.Factory
+	configAccess      clientcmd.ConfigAccess
+	KubernetesClient  corev1client.CoreV1Interface
+}
+
+// NewAgentOptions creates a new *AgentOptions with sane defaults
+func NewAgentOptions() *AgentOptions {
+	return &AgentOptions{}
+}
+
+func NewCmdCreateAgent(configuration *config.Configuration) *cobra.Command {
+
+	o := NewAgentOptions()
+
+	cmd := &cobra.Command{
+		Use:     "agent",
+		Aliases: []string{},
+		Short:   agentShort,
+		Long:    agentLong,
+		Example: agentExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			if err := o.Complete(configuration); err != nil {
+				return err
+			}
+
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			if err := o.Run(); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+func (o *AgentOptions) Complete(cfg *config.Configuration) error {
+
+	o.configAccess = clientcmd.NewDefaultPathOptions()
+	o.configuration = cfg
+
+	ac := configuration.NewClient(cfg)
+	o.ArmoryClient = ac
+
+	f := o.getKubernetesFactory()
+	o.kubernetesFactory = f
+
+	kc, err := o.getKubernetesClient()
+	if err != nil {
+		return err
+	}
+	o.KubernetesClient = kc
+
+	contextNames, err := o.getContexts()
+	if err != nil {
+		return err
+	}
+	o.contextNames = contextNames
+	o.Context = o.ArmoryClient.ArmoryCloudClient.Context
+	return nil
+}
+
+// Run performs the execution of 'create agent' sub command
+func (o *AgentOptions) Run() error {
+	promptSelectAgent := promptui.Select{
+		Label:  "Please select a context. Your agent will be deployed into the cluster you choose",
+		Items:  o.contextNames,
+		Stdout: &util.BellSkipper{},
+	}
+
+	_, requestedContext, err := promptSelectAgent.Run()
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to select a context to deploy to; %v\n", err))
+	}
+
+	err = o.useContext(requestedContext)
+	if err != nil {
+		return errors.New(fmt.Sprintf("failed to set context %s; %v\n", requestedContext, err))
+	}
+
+	ctx, cancel := context.WithTimeout(o.Context, time.Minute)
+	defer cancel()
+
+	// fetch the list of agents
+	existingAgents, err := o.ArmoryClient.Agents().List(ctx)
+	if err != nil {
+		return err
+	}
+
+	agentNameAlreadyExistFunc := func(name string) bool {
+		_, exists := lo.Find(existingAgents, func(a model.Agent) bool {
+			return name == a.AgentIdentifier
+		})
+		return exists
+	}
+
+	// set agent name
+	promptSetAgentName := promptui.Prompt{
+		Label: fmt.Sprintf("Provide an agent identifier %s", lo.Ternary(agentNameAlreadyExistFunc(requestedContext), "", fmt.Sprintf("[default=%s]", requestedContext))),
+		Validate: func(name string) error {
+			if agentNameAlreadyExistFunc(name) {
+				return errors.New("sorry, there's already an agent with that name in your tenant")
```

Comment:
The armory go style is to have these be defined as vars at the top of the package so that a non-technical person could easily edit and improve them.

```go
var (
  ErrDuplicateAgent = errors.New(...)
)
```